salt/redmine/init.sls

```diff
@@ -0,0 +1,88 @@
+{% from 'lib.sls' import apache, create_user %}
+
+include:
+  - apache
+  - apache.modules.passenger
+  - mysql
+  - rvm
+
+{{ create_user(pillar.redmine.user) }}
+
+/home/{{ pillar.redmine.user }}/public_html:
+  file.directory:
+    - user: {{ pillar.redmine.user }}
+    - group: {{ pillar.redmine.user }}
+    - require:
+      - user: {{ pillar.redmine.user }}_user_exists
+
+redmine dependencies:
+  pkg.installed:
+    - pkgs:
+      - subversion
+      - ruby-dev
+      - libncurses5-dev
+      - libmysqlclient-dev
+      - ghostscript
+      - imagemagick
+    - require:
+      - pkg: libapache2-mod-passenger
+      - rvm: ruby-{{ pillar.ruby_version }}
+
+# Deploy redmine code 
+redmine deploy:
+  svn.latest:
+    - name: https://svn.redmine.org/redmine/branches/{{ pillar.redmine.svn.branch }}
+    - target: /home/{{ pillar.redmine.user }}/public_html
+    - rev: {{ pillar.redmine.svn.revision }}
+    - require:
+      - pkg: redmine dependencies
+      - file: /home/{{ pillar.redmine.user }}/public_html
+
+set redmine directory permissions:
+  file.directory:
+    - names:
+      - /home/{{ pillar.redmine.user }}/public_html/tmp
+      - /home/{{ pillar.redmine.user }}/public_html/files
+      - /home/{{ pillar.redmine.user }}/public_html/log
+      - /home/{{ pillar.redmine.user }}/public_html/public/plugin_assets
+    - user: {{ pillar.redmine.user }}
+    - group: {{ pillar.redmine.user }}
+    - mode: 755
```

Comment:
What is their mode without this? I thought directories defaulted to 755.

Reply:
I believe you are correct, 755 should be the default but permissions can easily change.
I quite like this as a sanity check

Reply:
Makes sense. I've added the link to the file.